ansible/roles/cuda/tasks/main.yaml

```diff
@@ -71,3 +79,65 @@
     state: present
     create: true
     mode: 0644
+
+- name: Create Vulkan directory
+  become: true
+  ansible.builtin.file:
+    path: /etc/vulkan/icd.d
+    state: directory
+    mode: "0755"
+
+- name: Create OpenGL directory
+  become: true
+  ansible.builtin.file:
+    path: /etc/glvnd/egl_vendor.d
+    state: directory
+    mode: "0755"
+
+- name: Create OpenCL directory
+  become: true
+  ansible.builtin.file:
+    path: /etc/OpenCL/vendors
+    state: directory
+    mode: "0755"
+
+- name: Register Vulkan GPU vendors
+  become: true
+  ansible.builtin.get_url:
+    url: https://gitlab.com/nvidia/container-images/vulkan/raw/dc389b0445c788901fda1d85be96fd1cb9410164/nvidia_icd.json
+    dest: /etc/vulkan/icd.d/nvidia_icd.json
+    mode: "0644"
+
+- name: Set permissions for Vulkan GPU vendors
+  become: true
+  ansible.builtin.file:
+    path: /etc/vulkan/icd.d/nvidia_icd.json
+    mode: "0644"
+
+- name: Register OpenGL GPU vendors
+  become: true
+  ansible.builtin.get_url:
+    url: https://gitlab.com/nvidia/container-images/opengl/raw/5191cf205d3e4bb1150091f9464499b076104354/glvnd/runtime/10_nvidia.json
+    dest: /etc/glvnd/egl_vendor.d/10_nvidia.json
+    mode: "0644"
+
+- name: Set permissions for OpenGL GPU vendors
+  become: true
+  ansible.builtin.file:
+    path: /etc/glvnd/egl_vendor.d/10_nvidia.json
+    mode: "0644"
+
+- name: Register OpenCL GPU vendors
+  become: true
+  ansible.builtin.file:
+    path: /etc/OpenCL/vendors/nvidia.icd
+    state: touch
+    mode: "0644"
+
+- name: Set permissions for OpenCL GPU vendors
+  become: true
+  ansible.builtin.lineinfile:
+    path: /etc/OpenCL/vendors/nvidia.icd
+    line: libnvidia-opencl.so.1
+    create: true
+    mode: "0644"
```

Comment:
Why were these necessary?

These didn't exist in the previous role or the installation instructions.

Are these also necessary in the source installation? (If they are not, we should add this information to the readme too.)

Reply:
@oguzkaganozt thanks for the updates, could you inform me about this part too?

Reply:
Those steps are currently at https://github.com/autowarefoundation/autoware/blob/e6dc52c0a3b7fec293a95d9889f1433b8b927016/docker/autoware-universe/Dockerfile#L47

If the question is why are they moved to a task, I assume it relates to https://github.com/autowarefoundation/autoware/pull/4072#discussion_r1457265489. If the question is why they are needed at all, see https://github.com/autowarefoundation/autoware/pull/2736

EDIT: I think at the moment that's only needed for TVM inference. But given the negligible size of the files, I don't think it hurts to install them all the time.

Reply:
I see, I understand the need for the single source of truth. Thanks for the explanations.

Just to have it on record, on my machine, I didn't run these commands before yet I have:

```
/etc/vulkan/icd.d
/etc/glvnd/egl_vendor.d
/etc/OpenCL/vendors
```
these folders.

And for the icd files:
```
# These don't exist on my machine:
/etc/vulkan/icd.d/nvidia_icd.json
/etc/glvnd/egl_vendor.d/10_nvidia.json

# This exists on my machine:
/etc/OpenCL/vendors/nvidia.icd
# and its contents are same with the one in the ansible script.
```

> EDIT2: Oh, that's the CUDA task. It doesn't look right, Vulkan is orthogonal to CUDA.

They are all Nvidia related at least. Maybe we can move them to the `nvidia_container_toolkit` task, what do you think?

Reply:
> They are all Nvidia related at least.

True, I missed that. I don't think anybody is trying to enable other vendors at the moment anyway.

> Maybe we can move them to the nvidia_container_toolkit task, what do you think?

Isn't nvidia_container_toolkit is for installation on the host? Whereas those files get installed into the runtime environment. I think it is best to keep it here.

Reply:
You're right, OK, let's keep it here then.